src/freenet/config/ConfigCallback.java

```diff
@@ -3,7 +3,54 @@
  * http://www.gnu.org/ for further details of the GPL. */
 package freenet.config;
 
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
 public abstract class ConfigCallback<T> {
+
+	/**
+	 * Create a config callback from lambdas. Needs to be specialized
+	 * in Subtypes of ConfigCallback to match the nodeConfig.register API.
+	 *
+	 * @param set accepts the new value and return an Exception or null (if no exception)
+	 */
+	public static <T> ConfigCallback<T> from(Supplier<T> get, Function<T, Exception> set) {
```

Comment:
I get why you’re doing this but functions returning exceptions rub me all kinds of wrong ways. I’d prefer something like this:

```java
public interface ConfigConsumer<T> {
    void accept(T value) throws InvalidConfigValueException, NodeNeedRestartException;
}
```

This way we don’t need casting in the `set()` method, and we don’t have to return anything from the `Consumer`.

Reply:
that’s much nicer — thank you!